### PR TITLE
Transceiver renaming 

### PR DESCRIPTION
**Pull request by bentoi**
This PR renames the various transceiver classes to instead use the suffix "socket" instead of "transceiver".

- ITransceiver is now an abstract SingleStreamSocket class
- MultiStreamTransceiver is now MultiStreamSocket
- MultiStreamTransceiverWithUnderlyingTransceiver is now MultiStreamOverSingleStreamSocket
- LegacyTransceiver is now Ice1NetworkSocket
- Other XxxTransceiver classes are now XxxSocket
- TransceiverStream is now SocketStream
- Other XxxStream classes remain named XxxStream (SlicStream, Ice1NetworkStream, ColocatedStream)

### CODE REVIEW

File: csharp/src/Ice/SingleStreamSocket.cs [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+using System.Collections.Generic;
+using System.Net.Security;
+using System.Net.Sockets;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace ZeroC.Ice
+{
+    /// <summary>A single-stream socket enables transmitting and receiving raw binary data over a transport such
+    /// as TCP, UDP, TLS or WebSocket. More transports can be supported by deriving this class.</summary>
```
Comment (bernardnormier) [inline csharp/src/Ice/SingleStreamSocket.cs:13]: I would remove TLS from this list.

File: csharp/src/Ice/Ice1NetworkSocketStream.cs [hunk 2]
```diff
@@ -7,13 +7,13 @@
 
 namespace ZeroC.Ice
 {
-    /// <summary>The LegacyStream class provides a stream implementation of the LegacyTransceiver and Ice1 protocol.
-    /// </summary>
-    internal class LegacyStream : SignaledTransceiverStream<(Ice1Definitions.FrameType, ArraySegment<byte>)>
+    /// <summary>The Ice1NetworkSocketStream class provides a stream implementation of the Ice1NetworkSocketSocket and
```
Comment (bernardnormier) [inline csharp/src/Ice/Ice1NetworkSocketStream.cs:10]: `Ice1NetworkSocket`

File: csharp/src/Ice/MultiStreamSocket.cs [hunk 3]
```diff
@@ -12,17 +12,22 @@
 
 namespace ZeroC.Ice
 {
-    /// <summary>The MultiStreamTransceiver abstract base class to implement multi-stream transports.</summary>
-    public abstract class MultiStreamTransceiver : IDisposable
+    /// <summary>A multi-stream socket enables transmitting raw binary data over multiple independent streams. The
```
Comment (bernardnormier) [inline csharp/src/Ice/MultiStreamSocket.cs:15]: Would it be accurate to describe a MultiStreamSocket (and a SingleStreamSocket) as an object associated with an established connection? Like the local end of a connection?

**Review by bernardnormier — APPROVED** [review]
Looks good!

File: csharp/src/Ice/MultiStreamSocket.cs (hunk 3 above)
Comment (bentoi) [reply csharp/src/Ice/MultiStreamSocket.cs:15]: Sure.  So are you suggesting to fix the comment? Something like `A XXX-stream socket represents the local end of a network connection and enables transmitting raw binary data over ...` or do you have a better suggestion?

Comment (bentoi) [reply csharp/src/Ice/MultiStreamSocket.cs]: It nows says: `A single-stream socket represents the local end of a network connection and enables transmitting raw binary data over a transport such as TCP, UDP or WebSocket.`

**Pull request merged**